src/targets/cpu/include/migraphx/cpu/pointwise.hpp

```diff
@@ -43,7 +43,8 @@ struct multi_index
     {
         assert(n < max_size);
         std::copy(s.lens().begin(), s.lens().end(), dims);
-        s.multi_copy(i, index, index + max_size);
+        const std::vector<size_t> m_idx = s.multi(i);
+        std::copy(m_idx.begin(), m_idx.end(), index);
```

Comment:
This should use `multi_copy` so we dont need to construct a temporary vector(which uses heap allocation). The `index` member is an array so we dont need to do heap allocation when constructing.

Reply:
Added back.